plugins/inmemoryscanner/src/lib/Scanner.cpp

```diff
@@ -13,20 +13,23 @@ using VmiCore::Plugin::PluginInterface;
 
 namespace InMemoryScanner
 {
-    Scanner::Scanner(const PluginInterface* pluginInterface,
+    Scanner::Scanner(PluginInterface* pluginInterface,
                      std::shared_ptr<IConfig> configuration,
                      std::unique_ptr<YaraInterface> yaraEngine,
                      std::unique_ptr<IDumping> dumping)
         : pluginInterface(pluginInterface),
           configuration(std::move(configuration)),
           yaraEngine(std::move(yaraEngine)),
           dumping(std::move(dumping)),
-          logger(this->pluginInterface->newNamedLogger(INMEMORY_LOGGER_NAME)),
-          inMemResultsLogger(this->pluginInterface->newNamedLogger(INMEMORY_LOGGER_NAME))
+          logger(pluginInterface->newNamedLogger(INMEMORY_LOGGER_NAME)),
+          inMemResultsLogger(pluginInterface->newNamedLogger(INMEMORY_LOGGER_NAME))
     {
         logger->bind({{VmiCore::WRITE_TO_FILE_TAG, LOG_FILENAME}});
         inMemResultsLogger->bind(
             {{VmiCore::WRITE_TO_FILE_TAG, (this->configuration->getOutputPath() / TEXT_RESULT_FILENAME).string()}});
+
+        pluginInterface->registerProcessTerminationEvent([this](auto&& a)
+                                                         { scanProcess(std::forward<decltype(a)>(a)); });
```

Comment:
Why is this different to the usage in ApiTracing? Both callbacks should take the same parameter type, right?